### PR TITLE
fix: search - dropdown focus

### PR DESCRIPTION
**Pull request by AdamPawlinski**
# Related issue

Closes [#1086](https://vsf.atlassian.net/browse/SFUI2-1086)

# Scope of work

Added separate ref to handle focus trap. It allows to handle it and dropdown floating as well. 

# Screenshots of visual changes

search

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

File: packages/sfui/frameworks/vue/composables/useTrapFocus/useTrapFocus.ts [hunk 1]
```diff
@@ -42,14 +42,14 @@ export const useTrapFocus = (containerElementRef: Ref<HTMLElement | undefined>,
   };
   const currentlyFocused = ref<HTMLElement | undefined>();
   const focusableElements = ref<FocusableElement[]>([]);
-  let containeHTMLElement: HTMLElement | undefined;
+  let containerHTMLElement: HTMLElement | undefined;
```
Comment (Szymon-dziewonski) [inline packages/sfui/frameworks/vue/composables/useTrapFocus/useTrapFocus.ts:45]: 👍

File: apps/preview/next/pages/showcases/Search/SearchBasic.tsx [hunk 2]
```diff
@@ -138,14 +143,17 @@ export default function SearchBasic() {
         }
       />
       {isOpen && (
-        <div ref={refs.setFloating} style={style} className="right-0 left-0">
+        <div ref={refs.setFloating} style={style} className="left-0 right-0">
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/Search/SearchBasic.tsx:146]: suggestion: please do remove headwing, this plugin for tailwind ordering class, there was once discussion about it, I removed it and for now IMHO we should not have something like that, I mean if we will have extensions like that we should have it all or not at all. But then we need to document it etc

**Review by Szymon-dziewonski — APPROVED** [review]
good job!

File: apps/preview/next/pages/showcases/Search/SearchBasic.tsx (hunk 2 above)
Comment (AdamPawlinski) [reply apps/preview/next/pages/showcases/Search/SearchBasic.tsx:146]: Ok, I removed it, but I'll leave this change like it is, if it's okay?

Comment (Szymon-dziewonski) [reply apps/preview/next/pages/showcases/Search/SearchBasic.tsx:146]: Sure, we can keep it, thank you :)

**Pull request merged**